src/API/Recruit.Api/Queries/GetLiveVacanciesQueryHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore;
+using MediatR;
+using SFA.DAS.Recruit.Api.Helpers;
+using SFA.DAS.Recruit.Api.Models;
+
+namespace SFA.DAS.Recruit.Api.Queries;
+
+public class GetLiveVacanciesQueryHandler : IRequestHandler<GetLiveVacanciesQuery, GetLiveVacanciesQueryResponse>
+{
+    private readonly IQueryStoreReader _queryStoreReader;
+
+    public GetLiveVacanciesQueryHandler(IQueryStoreReader queryStoreReader)
+    {
+        _queryStoreReader = queryStoreReader;
+    }
+
+    public async Task<GetLiveVacanciesQueryResponse> Handle(GetLiveVacanciesQuery request, CancellationToken cancellationToken)
+    {
+        var liveVacanciesToFetch = request.PageSize * request.PageNumber;
+
+        var queryResult = liveVacanciesToFetch > 1000 ? await _queryStoreReader.GetAllLiveVacancies(1000) : await _queryStoreReader.GetAllLiveVacancies(liveVacanciesToFetch);
```

Comment:
I think this will fail when i have a page size of 1000 and request anything over page 1